delivery/client.py

```diff
@@ -117,6 +124,37 @@ def upload_metadata(
 
         res.raise_for_status()
 
+    def component_metadata(
+        self,
+        name: str,
+        type: str = None,
+        select: str = None,
+        version: str = None,
+    ) -> list[dso.model.ArtefactMetadata]:
+        url = self._routes.component_metadata()
+
+        # No validation done here. It is left to the caller to ensure that all parameters are
```

Comment:
it would be nice to add a docstr explaining the parameters, or at least referencing where such a documentation can be found (esp. `select` is probably not obvious)